### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by billionsjoel**
**Project requirements**

  - Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)[](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#interface)[](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#turn-your-person-class-to-nameable)[](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#prepare-base-decorator)[](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#prepare-capitalizedecorator-and-trimmerdecorator)[](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md#see-your-decorators-in-action)
    - Think about how you can use two decorators in order to capitalize and trim people's names.
    - In the following steps you will get instructions for implementing the Decorator design pattern for this case.
- Interface
  - Create a class Nameable.
  - Implement a method called correct_name that will raise a NotImplementedError.
  - Turn your Person class to Nameable
  - Make sure that your Person class inherits from Nameable
  - Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- Prepare base Decorator
  - Make sure that it inherits from Namable.
  - In the constructor assign a namable object from params to an instance variable.
  - Implement the correct_name method that returns the result of the correct_name method of the @namable.
- Prepare CapitalizeDecorator and TrimmerDecorator
  - For the CapitalizeDecorator:
  - Create a class that inherits from the base Decorator class.
  - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
  - Create a class that inherits from the base Decorator class.
  - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```
Comment (Cathella) [inline base_decorator.rb:6]: - [x] Before the initialize method, please declare an `attr_accessor :nameable`

```
...
    attr_accessor :nameable
  
    def initialize(nameable)
      @nameable = nameable
    end
....
```

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0, 10]
```
Comment (Cathella) [inline trimmer_decorator.rb]: - [x] This looks good! But can also use a straight forward conditional statement like this which will be great for an array, string, etc

`@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

**Review by Cathella — CHANGES_REQUESTED** [review]
## 🟡 Status: Changes Required 🛑

### Hi @billionsjoel 👋🏻, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## To Highlight 👏🏻

✅ No linter errors.
✅ You used correct Github Flow.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 2 above)
Comment (billionsjoel) [reply trimmer_decorator.rb]: Hello @Cathella ,

Thank you for the refactor. This is a better way. 👍

**Review by Kingstalux — APPROVED** [review]
Hi @billionsjoel,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966

Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**